jxl/src/frame.rs

```diff
@@ -190,6 +256,27 @@ impl Frame {
 
         Ok(())
     }
+
+    pub fn finalize(mut self) -> Result<Option<DecoderState>> {
+        if self.header.can_be_referenced {
+            // TODO(firsching): actually decode the image here, instead of setting it
```

Comment:
Probably the image has already been decoded by this point, in the long run... It probably should not be decoded here at all

Reply:
changed